File: common/src/main/java/com/ssblur/scriptor/word/descriptor/power/SolarPowerDescriptor.java
package com.ssblur.scriptor.word.descriptor.power;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.api.word.descriptor.StrengthDescriptor;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.api.word.descriptor.CastDescriptor;


public class SolarPowerDescriptor extends Descriptor implements CastDescriptor, StrengthDescriptor {
  @Override
  public Cost cost() { return new Cost(0, COSTTYPE.ADDITIVE_POST); }

  @Override
  public boolean cannotCast(Targetable caster) {
    return !caster.getLevel().isDay();
  }

  @Override
  public double strengthModifier() {
    return 1.5;
  }

  @Override
  public boolean allowsDuplicates() {
    return false;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/power/SolarPowerDescriptor.kt
package com.ssblur.scriptor.word.descriptor.power

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.descriptor.CastDescriptor

class SolarPowerDescriptor : Descriptor(), CastDescriptor, StrengthDescriptor {
    override fun cost(): Cost {
        return Cost(0.0, COSTTYPE.ADDITIVE_POST)
    }

    override fun cannotCast(caster: Targetable?): Boolean {
        return !caster!!.level.isDay
    }

    override fun strengthModifier(): Double {
        return 1.5
    }

    override fun allowsDuplicates(): Boolean {
        return false
    }
}
